### PR TITLE
Added Sorting Manager

### PR DESCRIPTION
**Pull request by alexdev03**
The system allows input of numerical values to generate a team name.

### CODE REVIEW

File: src/main/java/net/william278/velocitab/player/TabPlayer.java [hunk 1]
```diff
@@ -126,18 +126,10 @@ public CompletableFuture<String> getNametag(@NotNull Velocitab plugin) {
...
-                .thenApply(truncated -> truncated + getPlayer().getUniqueId().toString().substring(0, 4)) // Make unique
-                .thenApply(teamName -> this.teamName = teamName)
-                .exceptionally(e -> {
-                    plugin.log(Level.ERROR, "Failed to get team name for " + player.getUsername(), e);
-                    return "";
+        return plugin.getSortingManager().getTeamName(this)
+                .thenApply(teamName -> {
+                    this.teamName = teamName;
+                    return teamName;
                 });
```
Comment (WiIIiam278) [inline src/main/java/net/william278/velocitab/player/TabPlayer.java:133]:
```suggestion
        return plugin.getSortingManager().getTeamName(this)
                .thenApply(teamName -> {
                    return this.teamName = teamName;
                });
```

Use Optional#map.orElse if you make getSortingManager return an Optional

File: src/main/java/net/william278/velocitab/player/Role.java [hunk 2]
```diff
@@ -73,29 +73,9 @@ public Optional<String> getSuffix() {
 
     @NotNull
     protected String getWeightString() {
-        return compressNumber(Integer.MAX_VALUE / 4d - weight);
+        return String.valueOf(weight);
```
Comment (WiIIiam278) [inline src/main/java/net/william278/velocitab/player/Role.java:76]: `Integer.toString(weight)`

File: src/main/java/net/william278/velocitab/Velocitab.java [hunk 3]
```diff
@@ -170,6 +173,14 @@ private void disableScoreboardManager() {
         }
     }
 
+    private void prepareSortingManager() {
+        this.sortingManager = new SortingManager(this);
+    }
+
+    public SortingManager getSortingManager() {
```
Comment (WiIIiam278) [inline src/main/java/net/william278/velocitab/Velocitab.java:180]: Either annotate as @NotNull or make it return an optional (as per previous comment)

File: src/main/java/net/william278/velocitab/Velocitab.java [hunk 4]
```diff
@@ -170,6 +173,14 @@ private void disableScoreboardManager() {
         }
     }
 
+    private void prepareSortingManager() {
+        this.sortingManager = new SortingManager(this);
```
Comment (WiIIiam278) [inline src/main/java/net/william278/velocitab/Velocitab.java:177]: Probably no need to instantiate this class if sorting is disabled, right?

File: src/main/java/net/william278/velocitab/sorting/SortingManager.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            charList.add(0, digit);
+
+            wholePart /= Character.MAX_VALUE;
+        }
+
+        if (charList.isEmpty()) {
+            charList.add((char) 0);
+        }
+
+        return charList.stream().map(String::valueOf).collect(Collectors.joining()) + decimalChar;
```
Comment (WiIIiam278) [inline src/main/java/net/william278/velocitab/sorting/SortingManager.java:102]: So I don't forget, can we remove some of the newlines here?

**Review by WiIIiam278 — CHANGES_REQUESTED** [review]
Just a few nits, nice work :+1:

File: src/main/java/net/william278/velocitab/Velocitab.java (hunk 4 above)
Comment (alexdev03) [reply src/main/java/net/william278/velocitab/Velocitab.java:177]: I would keep it, so if someone reloads the plugin it's ready to use.

File: src/main/java/net/william278/velocitab/Velocitab.java (hunk 3 above)
Comment (alexdev03) [reply src/main/java/net/william278/velocitab/Velocitab.java:180]: I would go with NotNull

File: src/main/java/net/william278/velocitab/Velocitab.java [hunk 6]
```diff
@@ -170,6 +173,14 @@ private void disableScoreboardManager() {
...
+    }
+
+    public @NotNull SortingManager getSortingManager() {
```
Comment (WiIIiam278) [inline src/main/java/net/william278/velocitab/Velocitab.java:180]:
```suggestion
    @NotNull
    public SortingManager getSortingManager() {
```

**Review by WiIIiam278 — APPROVED** [review]

**Pull request merged**